processor/src/main/java/org/mapstruct/ap/writer/IndentationCorrectingWriter.java

```diff
@@ -142,27 +162,113 @@ void onBufferFinished(StateContext context) throws IOException {
                 flush( context );
             }
 
-            private void flush(StateContext context) throws IOException {
-                if ( null != context.characters && context.currentIndex - context.lastStateChange > 0 ) {
-                    context.writer.write(
-                        context.characters,
-                        context.lastStateChange,
-                        context.currentIndex - context.lastStateChange
-                    );
+        },
 
-                    if ( DEBUG ) {
-                        System.out.print(
-                            new String(
-                                java.util.Arrays.copyOfRange(
-                                    context.characters,
-                                    context.lastStateChange,
-                                    context.currentIndex
-                                )
-                            )
-                        );
-                    }
+        /**
+         * In a String definition, Between un-escaped quotes " "
+         */
+        IN_TEXT_IN_STRING {
```

Comment:
If it would be a classical state pattern (abstract base class, each state is a class), then IN_STRING would be a substate of IN_TEXT (inherits from).. I now tried to express that in the name.